reps/2023-05-10-actors-batch-remote-api.md

```diff
@@ -0,0 +1,204 @@
+## Summary
+### General Motivation
+**Core Motivation**:
+1. Improve the performance of batch calling ActorTask.
+2. Implement Ray's native collective communication library through this interface.
+
+
+In distributed computing scenarios, such as big data computing、AI training and inference, there is a need to send a large number of RPC requests(ActorTask) to multiple Actors in batches.  
+For example, in a typical star-topology architecture with one master Actor and 400 worker Actors, the same computational request needs to be sent to 400 worker Actors. 
+However, the computing tasks of each Worker Actor are very short. In such a scenario, the performance requirements for executing batch actor task remote of a large number of Actor are very high.  
+Therefore, for the scenario of batch calling Actor tasks, I want to add a new optimization API, batch_remote(), to improve the performance of batch submission of Actor Task calls.   
+After my own performance testing and comparison, this interface has significant performance improvements.
+
+Current situation of batch calling actor tasks:
+```
+actors = [WorkerActor.remote() for _ in range(400)]
+
+# This loop's repeated invocation actually wastes a lot of performance.
+for actor in actors:
+  actor.compute.remote(args)
+```
+
+Using the new Batch Remote API:
+```
+actors = [WorkerActor.remote() for _ in range(400)]
+
+# Calling it only once can greatly improve performance.
+ray.batch_remote(actors).compute.remote(args)
+```
+
+The Batch Remote API can save the following performance costs(The N is the number of Actors):  
+1. Reduce (N-1) times of parameter serialization performance time.
+2. Reduce (N-1) times of putting parameter into object store performance time for scenarios with large parameters.
+3. Reduce (N-1) times of python and C++ execution layer switching and repeated parameter verification performance time.
+
+
+### Should this change be within `ray` or outside?
+This requires adding a new interface in Ray Core.
```

Comment:
I believe this should be added as an experimental & internal API first:
```python
ray.experimental._batch_remote(actors).compute.remote(args)
```

Reply:
Yes. At first, it should be placed in the experimental module.